python/equistore-operations/equistore/operations/_utils.py

```diff
@@ -198,8 +198,6 @@ def _check_same_gradients_impl(
     for parameter, grad_a in a.gradients():
         grad_b = b.gradient(parameter)
 
-        _check_same_gradients(grad_a, grad_b, props=props, fname=fname)
-
```

Comment:
@agoscinski I do not see any reasons why this was here. It should have created an infinite recursion loop, since 
`_check_same_gradients` will call `_check_same_gradients_impl` which is the current function...

Reply:
I think because it was already there before PR #280 and we just did not notice it when doing the refactor

See one commit before:
https://github.com/lab-cosmo/equistore/blob/8eb034f69e1c968fdb562cadda41301a5c7d5190/python/equistore-operations/equistore/operations/_utils.py